### ISSUE TITLE
Loading cpp highlighter alone causes an error

### ISSUE THREAD

**Issue by SilentAntenna** [highlightjs/highlight.js#2670]
The current cpp highlighter is basically a redirection to c-like highlighter. If one registers this highlighter as in the documentation:

```
const hljs = require("highlight.js/lib/core");
// separately require languages
hljs.registerLanguage('cpp', require('highlight.js/lib/languages/cpp'));

const highlightedCode = hljs.highlight('cpp', 'cout << "Hello World!"').value
```

Without loading c-like highlighter first, the script above will throw an error.

If this behavior is intended, it should be mentioned in the documentation.

**Comment by joshgoebel** [comment]
> If this behavior is intended, it should be mentioned in the documentation.

Where in the docs do you think this should be?  Is the error thrown not clear enough?  

In version 11 we may just switch to static linking to avoid this completely.

**Comment by SilentAntenna** [comment]
The error itself is not very clear. One would only see something like "accessing property XXX of undefined" and they need to look at the source code to find out why.

It's not a big deal. But it is still a little bit *surprising* as the library doesn't work exactly as expected.

Maybe we should state in the "Custom Initialization" section that highlighters of different languages may rely on each other, and loading only the highlighter for the target language is not enough. At least for the current version.

**Comment by joshgoebel** [comment]
Well, that is incorrect - the error at least should be quire nice.  Fixed in be8348c85854ea09077ad118cf1b11b36b5fba35.

**Comment by joshgoebel** [comment]
@SilentAntenna Would that resolve this for you?

**Comment by SilentAntenna** [comment]
Yes. Now the user will know what the problem is. Thanks.

**Comment by joshgoebel** [comment]
It's merged, but there is no release yet.  Closing.  Will be in the next release.

**Comment by jacobedawson** [comment]
For googlers, especially those using React & module imports getting the "language definition for cpp could not be registered, the c-like language is required but not loaded" error, I solved the issue by importing c-like and registering both c & c++ (cpp) as clike:

```
import hljs from "highlight.js/lib/core";
import clike from "highlight.js/lib/languages/c-like";
hljs.registerLanguage("c", clike);
hljs.registerLanguage("cpp", clike);
```

In the referenced fix you can see a comment: 
```
  // Until C is actually different than C++ there is no reason to auto-detect C
  // as it's own language since it would just fail auto-detect testing or	  // as it's own language since it would just fail auto-detect testing or
  // simply match with C++.	  // simply match with C++.
```

I also found this same issue previously with html, whereby I ended up importing xml and registering it as html, which is fine since it's a default export and we can name it whatever we want:

```
import html from "highlight.js/lib/languages/xml";
hljs.registerLanguage("html", html);
```

There might be better fixes but when trying to balance 1000 deps at once this does the trick for now.

**Comment by joshgoebel** [comment]
We're looking at how to solve this in the future, but yep it's an issue for now for anyone MANUALLY doing all the module loading.  If you build the library using the build tools or the website then we (of course) handle this for you.

>  I ended up importing xml and registering it as html

This also "just works" for many use cases if you register it as xml, since html is an alias of xml.

**Comment by joshgoebel** [comment]
```
import clike from "highlight.js/lib/languages/c-like";
hljs.registerLanguage("c", clike);
hljs.registerLanguage("cpp", clike);
```

This is also slightly broken and may not register the languages correctly - autodetect may not work, etc.  The correct path is to load all 3:

```
import `.../c-like`;
import `.../c`;
import `.../cpp`;
hljs.registerLanguage("c-like", ...
hljs.registerLanguage("c", ...
hljs.registerLanguage("cpp", ...
```

**Comment by joshgoebel** [comment]
@jacobedawson Please update because the way you're doing it WILL break in the future when we actually fix this. `c-like` is not intended to be used directly as it will disappear in the future - or become a "do nothing" just to not break requires.

**Comment by joshgoebel** [comment]
Actually I think I did it in a non-breaking way, but my comments and proper auto-detect and other things being slightly glitchy still apply.

Related: https://github.com/highlightjs/highlight.js/pull/2844

Unless there are issues I think we're finally going to fix this in 10.4.
